File: Strings/2549/2549.cpp
#include <iostream>
#include <unordered_map>
#include <sstream>
using namespace std;

void register_student(unordered_map<string, int>& SD, string year){
    string student_name;
    getline(cin, student_name);
    istringstream iss(student_name);
    string word, acronym;
    while(iss >> word){
        acronym += word[0];
    }
    SD[acronym + year]++;
}

int main(){
    int n;
    string year;
    while(cin >> n >> year){
        cin.ignore(); // <-- apenas aqui
        unordered_map<string, int> system_data;
        for(int i = 0; i < n; i++)
            register_student(system_data, year);
        int ans = 0;
        for (const auto& [key, value] : system_data) {
            if(value >= 2)
                ans += (value - 1);
        }
        cout << ans << endl;
    }
    return 0;
}

File: Strings/__2000__/2549/2549.py
def register_student(year) -> str:
    student_name = [word for word in input().split()]
    acronym = "".join([word[0] for word in student_name])
    return acronym + year
    
while True:
    try:
        n, year = [data for data in input().split()]
        system_data = {}
        for _ in range(int(n)):
            key = register_student(year)
            system_data[key] = system_data.get(key, 0) + 1
        ans = 0
        for value in system_data.values():
            if value >= 2: ans += (value - 1)
        print(ans)
    except EOFError: break